src/attr/_make.py

```diff
@@ -2762,4 +2786,36 @@ def pipe_converter(val):
 
         return val
 
+    if not PY2:
+        if not converters:
+            # If the converter list is empty, pipe_converter is the identity.
+            A = typing.TypeVar("A")
```

Comment:
nit: I don't like how it works with empty pipe, as it will put `TypeVar("A")` in `__init__` annotation. Can't figure a much better solution, though.